test/unit/org/apache/cassandra/cql3/ast/Conditional.java

```diff
@@ -77,29 +223,158 @@ public enum Kind
             }
         }
 
+        public final ReferenceExpression reference;
         public final Kind kind;
-        public final Reference reference;
 
         public Is(String symbol, Kind kind)
         {
-            this(Reference.of(Symbol.unknownType(symbol)), kind);
+            this(Symbol.unknownType(symbol), kind);
         }
 
-        public Is(Reference reference, Kind kind)
+        public Is(ReferenceExpression reference, Kind kind)
         {
-            this.kind = kind;
             this.reference = reference;
+            this.kind = kind;
         }
 
         @Override
-        public void toCQL(StringBuilder sb, int indent)
+        public void toCQL(StringBuilder sb, CQLFormatter formatter)
         {
-            reference.toCQL(sb, indent);
+            reference.toCQL(sb, formatter);
             sb.append(" IS ").append(kind.cql);
         }
     }
 
-    class Builder
+    class And implements Conditional
+    {
+        public final Conditional left, right;
+
+        public And(Conditional left, Conditional right)
+        {
+            this.left = left;
+            this.right = right;
+        }
+
+        @Override
+        public void toCQL(StringBuilder sb, CQLFormatter formatter)
+        {
+            left.toCQL(sb, formatter);
+            sb.append(" AND ");
+            right.toCQL(sb, formatter);
+        }
+
+        @Override
+        public Stream<? extends Element> stream()
+        {
+            return Stream.of(left, right);
+        }
+
+        @Override
+        public Conditional visit(Visitor v)
+        {
+            var u = v.visit(this);
+            if (u != this) return u;
+            Conditional left = this.left.visit(v);
+            Conditional right = this.right.visit(v);
+            if (this.left == left && this.right == right) return this;
+            return new And(left, right);
+        }
+
+        @Override
+        public List<Conditional> simplify()
+        {
+            List<Conditional> result = new ArrayList<>();
+            result.addAll(left.simplify());
+            result.addAll(right.simplify());
+            return result;
+        }
+    }
+
+    interface EqBuilder<T extends EqBuilder<T>>
+    {
+        T value(Symbol symbol, Expression e);
+        default T value(String symbol, int e)
+        {
+            return value(new Symbol(symbol, Int32Type.instance), Bind.of(e));
+        }
+    }
+
+    interface ConditionalBuilder<T extends ConditionalBuilder<T>> extends EqBuilder<T>
```

Comment:
rather than have each builder copy/paste and let them get out of sync, made these interfaces to define what you can do with a `Conditional`, now all builders are consistent.